src/index.ts

```diff
@@ -174,29 +175,54 @@ function initialSetup() {
 	logger('Main HTTP Server Complete.', 'info-quiet');
 
 	bonjour().publish({
-		name: 'tallyarbiter-'+currentConfig["uuid"],
+		name: 'tallyarbiter-'+(currentConfig["uuid"] || "unknown"),
 		type: 'tally-arbiter',
 		port: 4455,
 		txt: {
 			version: version,
-			uuid: currentConfig["uuid"]
+			uuid: (currentConfig["uuid"] || "unknown")
 		}
 	});
 	logger('TallyArbiter advertised over MDNS.', 'info-quiet');
 
 	logger('Starting socket.IO Setup.', 'info-quiet');
 
+	let tmpSocketAccessTokens: string[] = [];
 	io.sockets.on('connection', (socket) => {
 		const ipAddr = socket.handshake.address;
 
-		socket.on('login', (type: "settings" | "producer", username: string, password: string) => {
-			if((type === "producer" && username == currentConfig.security.username_producer && password == currentConfig.security.password_producer)
-			|| (type === "settings" && username == currentConfig.security.username_settings && password == currentConfig.security.password_settings)) {
-				//login successfull
-				socket.emit('login_result', true); //old response, for compatibility with old UI clients
-				socket.emit('login_response', { loginOk: true, message: "" });
-			} else {
-				//wrong credentials
+		const requireRole = (role: string) => {
+			return new Promise((resolve, reject) => {
+				console.log(tmpSocketAccessTokens);
+				if(typeof(tmpSocketAccessTokens[socket.id]) === undefined) {
+					let error_msg = "Access token required. Please login to use this feature.";
+					socket.emit('error', error_msg);
+					reject(error_msg);
+				}
+				let access_token = tmpSocketAccessTokens[socket.id];
+				validateAccessToken(access_token).then((user) => {
+					if(user.roles.includes(role) || user.roles.includes("admin")) {
+						resolve(user);
+					} else {
+						let error_msg = "Access denied. You are not authorized to do this.";
+					    socket.emit('error', error_msg);
+					    reject(error_msg);
+					}
+				}).catch((err) => {
+					socket.emit('error', err.message);
+					reject(err.message);
+				});
+			}); 
+		}
+
+		socket.on('login', (username: string, password: string) => {
+			authenticate(username, password).then((result) => {
+				console.log("result", result);
+                socket.emit('login_result', true); //old response, for compatibility with old UI clients
+				socket.emit('login_response', { loginOk: true, message: "", accessToken: result.access_token });
+			}).catch((error) => {
+				console.log("error", error);

```

Comment:
Maybe we make this logging a bit more useful by printing `User XYZ has attempted a login: wrong password` or simething like this.

Same for the `console.log` 4 lines above this one